Add stubs for Keyboard Lock API

xingri opened the pull request:
#### c1634333112e11d682186c93ed1c4d9972148678
<pre>
Add stubs for Keyboard Lock API
<a href="https://bugs.webkit.org/show_bug.cgi?id=265222">https://bugs.webkit.org/show_bug.cgi?id=265222</a>

Reviewed by NOBODY (OOPS!).

This patch is extracted from following WebPlatformForEmbedded WPEWebKit code changes:
&lt;<a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/0d7f83d8d30cf0561ebb5de57642d2a8372b91da">https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/0d7f83d8d30cf0561ebb5de57642d2a8372b91da</a>&gt;

co-authoured by:
Philippe Normand &lt;philn@webkit.org&gt;

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/keyboard-lock/Keyboard.cpp: Added.
(WebCore::Keyboard::create):
(WebCore::Keyboard::Keyboard):
(WebCore::Keyboard::navigator):
(WebCore::Keyboard::lock):
(WebCore::Keyboard::unlock):
* Source/WebCore/Modules/keyboard-lock/Keyboard.h: Added.
* Source/WebCore/Modules/keyboard-lock/Keyboard.idl: Added.
* Source/WebCore/Modules/keyboard-lock/KeyboardLockController.cpp: Added.
(WebCore::sharedController):
(WebCore::KeyboardLockController::shared):
(WebCore::KeyboardLockController::setSharedController):
* Source/WebCore/Modules/keyboard-lock/KeyboardLockController.h: Added.
* Source/WebCore/Modules/keyboard-lock/Navigator+Keyboard.idl: Added.
* Source/WebCore/Modules/keyboard-lock/NavigatorKeyboard.cpp: Added.
(WebCore::NavigatorKeyboard::NavigatorKeyboard):
(WebCore::NavigatorKeyboard::keyboard):
(WebCore::NavigatorKeyboard::from):
(WebCore::NavigatorKeyboard::supplementName):
* Source/WebCore/Modules/keyboard-lock/NavigatorKeyboard.h: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c1634333112e11d682186c93ed1c4d9972148678

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28092 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6733 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29442 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/30623 "Hash c1634333 for PR 20823 does not build (failure)") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25611 "Hash c1634333 for PR 20823 does not build (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/28589 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8749 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4119 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/30623 "Hash c1634333 for PR 20823 does not build (failure)") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28360 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5492 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24172 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/30623 "Hash c1634333 for PR 20823 does not build (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/4908 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/25169 "Found 2 new test failures: fast/dom/navigator-detached-no-crash.html, fast/ruby/ruby-overhang-crash.html (failure)") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/31312 "Hash c1634333 for PR 20823 does not build (failure)") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/24318 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/25699 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25610 "Found 1 new test failure: fast/dom/navigator-detached-no-crash.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/31312 "Hash c1634333 for PR 20823 does not build (failure)") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/27216 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4904 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3126 "Found 1 new test failure: fast/dom/navigator-detached-no-crash.html (failure)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/31312 "Hash c1634333 for PR 20823 does not build (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6444 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/34714 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5342 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7512 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5402 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/347318b6ceddfba13c851adbb659086dbc3bf491. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/20823

xingri commented:
Working on fixing build error through <https://github.com/WebKit/WebKit/pull/20827>.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e7b03edd57397cabda613abfcca1be74ffea51f3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/20823

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c24202f2410aa63f9939a2137cecaeaeafa705c7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/20823

emw-apple committed:
Add stubs for Keyboard Lock API
https://bugs.webkit.org/show_bug.cgi?id=265222

Reviewed by NOBODY (OOPS!).

This patch is extracted from following WebPlatformForEmbedded WPEWebKit code changes:
<https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/0d7f83d8d30cf0561ebb5de57642d2a8372b91da>

co-authoured by:
Philippe Normand <philn@webkit.org>

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/keyboard-lock/Keyboard.cpp: Added.
(WebCore::Keyboard::create):
(WebCore::Keyboard::Keyboard):
(WebCore::Keyboard::navigator):
(WebCore::Keyboard::lock):
(WebCore::Keyboard::unlock):
* Source/WebCore/Modules/keyboard-lock/Keyboard.h: Added.
* Source/WebCore/Modules/keyboard-lock/Keyboard.idl: Added.
* Source/WebCore/Modules/keyboard-lock/KeyboardLockController.cpp: Added.
(WebCore::sharedController):
(WebCore::KeyboardLockController::shared):
(WebCore::KeyboardLockController::setSharedController):
* Source/WebCore/Modules/keyboard-lock/KeyboardLockController.h: Added.
* Source/WebCore/Modules/keyboard-lock/Navigator+Keyboard.idl: Added.
* Source/WebCore/Modules/keyboard-lock/NavigatorKeyboard.cpp: Added.
(WebCore::NavigatorKeyboard::NavigatorKeyboard):
(WebCore::NavigatorKeyboard::keyboard):
(WebCore::NavigatorKeyboard::from):
(WebCore::NavigatorKeyboard::supplementName):
* Source/WebCore/Modules/keyboard-lock/NavigatorKeyboard.h: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c1634333112e11d682186c93ed1c4d9972148678. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/20823

philn commented:
This would need a rebase. Also could you had `ENABLE_KEYBOARD_LOCK` as experimental feature in cmake ports?

xingri commented:
Let me close this one since we decided to go with Fullscreen based API.
- https://github.com/WebKit/WebKit/pull/22842
